### PR TITLE
List repositories

### PR DESCRIPTION
**Pull request by alaibe**

### CODE REVIEW

File: spec/requests/users/repositories_spec.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    it 'returns a 412' do
+      get '/repositories'
+
+      expect(last_response.status).to be_eql 412
+    end
+  end
+
+  context 'When user corresponding to the api key does not exist' do
+    it 'returns a 403' do
+      get '/self', {}, { 'API_KEY' => 'random' }
```
Comment (kdisneur) [inline spec/requests/users/repositories_spec.rb]: Why `get '/self'` in the repositories spec?

File: dmc/customer/use_cases/list_repositories.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+module Customer
+  class ListRepositories < Struct.new(:user)
+
+    def run!
+      account     = user.find_account 'github'
+      repositories = GithubService.repositories account.oauth_token
+
+      repositories.map { |repository| RepositoryProvider.new(repository) }
```
Comment (kdisneur) [inline dmc/customer/use_cases/list_repositories.rb]: I don't understand the naming. What's a `RepositoryProvider`?

File: dmc/customer/services/github_service/octokit_github_service.rb [hunk 3]
```diff
@@ -13,5 +13,27 @@ def fetch_token(code)
...
+      client = Octokit::Client.new(access_token: oauth_token, auto_paginate: true)
+
+      user_repositories(client) + organization_repositories(client)
+    end
+
+    private
+
+    def user_repositories(client)
+      client.repositories.map do |repository|
+        { name: repository[:full_name], url: repository[:clone_url] }
```
Comment (kdisneur) [inline dmc/customer/services/github_service/octokit_github_service.rb]: To be sure to use the same columns across your "builders", you could maybe extract this line and the line 33 in another method?

File: spec/requests/users/repositories_spec.rb (hunk 1 above)
Comment (alaibe) [inline spec/requests/users/repositories_spec.rb]: copy/paste

File: dmc/customer/services/github_service/octokit_github_service.rb (hunk 3 above)
Comment (alaibe) [inline dmc/customer/services/github_service/octokit_github_service.rb]: :+1:

File: dmc/customer/use_cases/list_repositories.rb (hunk 2 above)
Comment (alaibe) [inline dmc/customer/use_cases/list_repositories.rb]: rename in remote project

**Pull request merged**